src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -231,16 +237,34 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
             !failSafe.MatchesFabricIndex(commandObj->GetAccessingFabricIndex()))
         {
             response.errorCode = CommissioningError::kInvalidAuthentication;
+            ChipLogError(FailSafe, "GeneralCommissioning: Got commissioning complete in invalid security context");
         }
         else
         {
+            if (failSafe.NocCommandHasBeenInvoked())
+            {
+                CHIP_ERROR err = fabricTable.CommitPendingFabricData();
```

Comment:
Do we have a sequencing problem with this commit happening here, but other things being committed or reverted in src/app/clusters/operational-credentials-server/operational-credentials-server.cpp after the event loop comes around the horn?  This doesn't seem right to me.  Shouldn't this all happen in one place?

Reply:
Where do you suggest it would happen. I want it to happen during the same event loop pass as the CommissioningComplete. I think it's safer like that, I can't see a different way off the top of my head that has zero races with the event loop since we are not doing any introspection of pending events.